src/stream.ts

```diff
@@ -281,54 +46,334 @@ export enum StreamStates {
+  transition ({ direction, flag }: StreamStateInput): [StreamStates, StreamStates] {
+    const prev = this.state
+
+    // return early if the stream is closed
+    if (this.state === StreamStates.CLOSED) {
+      return [prev, StreamStates.CLOSED]
     }
-    if (direction == 'inbound') {
+
+    if (direction === 'inbound') {
       switch (flag) {
         case pb.Message_Flag.FIN:
-          if (this.state == StreamStates.OPEN) {
-            this.state = StreamStates.READ_CLOSED;
-          } else if (this.state == StreamStates.WRITE_CLOSED) {
-            this.state = StreamStates.CLOSED;
+          if (this.state === StreamStates.OPEN) {
+            this.state = StreamStates.READ_CLOSED
+          } else if (this.state === StreamStates.WRITE_CLOSED) {
+            this.state = StreamStates.CLOSED
           }
-          break;
+          break
 
         case pb.Message_Flag.STOP_SENDING:
-          if (this.state == StreamStates.OPEN) {
-            this.state = StreamStates.WRITE_CLOSED;
-          } else if (this.state == StreamStates.READ_CLOSED) {
-            this.state = StreamStates.CLOSED;
+          if (this.state === StreamStates.OPEN) {
+            this.state = StreamStates.WRITE_CLOSED
+          } else if (this.state === StreamStates.READ_CLOSED) {
+            this.state = StreamStates.CLOSED
           }
-          break;
+          break
 
         case pb.Message_Flag.RESET:
-          this.state = StreamStates.CLOSED;
+          this.state = StreamStates.CLOSED
+          break
+
+        // no default
       }
     } else {
       switch (flag) {
         case pb.Message_Flag.FIN:
-          if (this.state == StreamStates.OPEN) {
-            this.state = StreamStates.WRITE_CLOSED;
-          } else if (this.state == StreamStates.READ_CLOSED) {
-            this.state = StreamStates.CLOSED;
+          if (this.state === StreamStates.OPEN) {
+            this.state = StreamStates.WRITE_CLOSED
+          } else if (this.state === StreamStates.READ_CLOSED) {
+            this.state = StreamStates.CLOSED
           }
-          break;
+          break
 
         case pb.Message_Flag.STOP_SENDING:
-          if (this.state == StreamStates.OPEN) {
-            this.state = StreamStates.READ_CLOSED;
-          } else if (this.state == StreamStates.WRITE_CLOSED) {
-            this.state = StreamStates.CLOSED;
+          if (this.state === StreamStates.OPEN) {
+            this.state = StreamStates.READ_CLOSED
+          } else if (this.state === StreamStates.WRITE_CLOSED) {
+            this.state = StreamStates.CLOSED
           }
-          break;
+          break
 
         case pb.Message_Flag.RESET:
-          this.state = StreamStates.CLOSED;
+          this.state = StreamStates.CLOSED
+          break
+
+        // no default
       }
     }
-    return [prev, this.state];
+    return [prev, this.state]
   }
 }
+
+export class WebRTCStream implements Stream {
+  /**
+   * Unique identifier for a stream
```

Comment:
why the funny way to comment instead of just `//`?

Reply:
Things are inconsistent and will be normalized in the next PR (this funny way will likely predominate, see https://jsdoc.app/)